authutils/utils.go

```diff
@@ -98,29 +95,18 @@ func HashSha256(data []byte) ([]byte, error) {
 	return hasher.Sum(nil), nil
 }
 
-// GetDefaultAccessTokenRequest returns a HTTP request to get an access token using a static token
-func GetDefaultAccessTokenRequest(host string, path string, token string) (*http.Request, error) {
-	if token == "" {
-		return nil, errors.New("service token is missing")
-	}
-
-	params := map[string]interface{}{
-		"auth_type": "static_token",
-		"creds": map[string]string{
-			"token": token,
-		},
-	}
-	data, err := json.Marshal(params)
-	if err != nil {
-		return nil, fmt.Errorf("error marshaling request body toget access token: %v", err)
+// StringOrNil returns a pointer to a non-empty string or nil if it matches nilCase
+func StringOrNil(v string, nilCase string) *string {
+	if v != nilCase {
+		return &v
 	}
+	return nil
+}
 
-	r, err := http.NewRequest("POST", host+path, bytes.NewReader(data))
-	if err != nil {
-		return nil, fmt.Errorf("error formatting request to get access token: %v", err)
+// StringOrDefault returns the contents of a string pointer or defaultVal if nil
+func StringOrDefault(v *string, defaultVal string) string {
```

Comment:
Both of these functions may become unnecessary once we fix the "all" vs. nil/"" issue.